apps/meteor/server/startup/migrations/v304.ts

```diff
@@ -0,0 +1,23 @@
+import { Settings } from '@rocket.chat/models';
+
+import { addMigration } from '../../lib/migrations';
+
+addMigration({
```

Comment:
we may need to come up with a different solution instead of using migrations..

we're not supposed to create new migrations between minor releases because they make the downgrade process (like in case of an issue appears) more difficult.. if the upgrade introduces a migration, it is not possible to downgrade without updating the DB record on database..

and since we'll be enforcing registration on every minor release update, we need a different solution, otherwise we'd need a new migration every release, which is far from ideal..

one solution we can explore is something similar to what is used for permissions:
https://github.com/RocketChat/Rocket.Chat/blob/75b3ac45c8917956feb96741e453e0131a41deee/apps/meteor/server/lib/migrations.ts#L295

let me know what you think and we can discuss about it.

Reply:
I think I understand perfectly your point here, let's talk about it for sure! I'm curious about the permissions solution! 😃